### PR TITLE
[APPS-3060] Update JDK version to 21

### PR DESCRIPTION
**Pull request by purusothaman-mm**
https://hyland.atlassian.net/browse/APPS-3060

### CODE REVIEW

File: samples/event-api-handlers/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM amazoncorretto:17-alpine3.18@sha256:6a3dc573328ced158ae25ffde792accbe8b1ff2e7b505c72d2d5e7b9bfa9ac26
+FROM amazoncorretto:21-alpine3.18@sha256:6a3dc573328ced158ae25ffde792accbe8b1ff2e7b505c72d2d5e7b9bfa9ac26
```
Comment (suneet-gupta) [inline samples/event-api-handlers/Dockerfile:1]: Please verify the image SHA

File: pom.xml [hunk 2]
```diff
@@ -113,7 +113,7 @@
     <project.scm.repository>alfresco-java-sdk</project.scm.repository>
     <project.year>2021</project.year>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <java.version>17</java.version>
+    <java.version>21</java.version>
```
Comment (suneet-gupta) [inline pom.xml:116]: Please check whether we can update java compile time as well  to 21

File: samples/extension-template/README.md [hunk 3]
```diff
@@ -14,13 +14,13 @@ to define a sample [```EventHandler```](../../alfresco-java-event-api/alfresco-j
 
 To properly build and run the project in a local environment it is required to have installed some tools.
 
-* Java 17:
```
Comment (SathishK-T) [inline samples/extension-template/README.md:17]: Need to update the java version in the other read file as well.

**Review by suneet-gupta — APPROVED** [review]

File: samples/event-api-handlers/Dockerfile (hunk 1 above)
Comment (purusothaman-mm) [reply samples/event-api-handlers/Dockerfile:1]: Confirmed with devops to use the same SHA image

File: samples/extension-template/README.md (hunk 3 above)
Comment (purusothaman-mm) [reply samples/extension-template/README.md:17]: Updated the java version to 21 in readme file

**Review by SathishK-T — APPROVED** [review]
LGTM

File: pom.xml (hunk 2 above)
Comment (purusothaman-mm) [reply pom.xml:116]: As discused confirming the changes with java 21 compile time as well

**Pull request merged**